### ISSUE TITLE
Send torrent to Synology DS fails following API deprecation

### ISSUE THREAD

**Issue by Namlaz** [SickGear/SickGear#1366]
#### Branch version commit hash: 0.25.18

Expectation | Result
------------ | -------------
Send torrent to Synology DS | Fails with misleading error

**Steps to reproduce:**
*1)* Upgrade Synology DS to 3.8.16-3566
*2)* Search for torrent in SickGear : fails to send to Syno DS.

<pre>2021-11-12 19:03:21 BACKLOG-1:78804 :: DownloadStation replied with: Max number of tasks reached
2021-11-12 19:03:21 BACKLOG-1:78804 :: DownloadStation: Unable to send torrent to client
2021-11-12 18:59:41 RECENT-SEARCH :: DownloadStation replied with: Max number of tasks reached
2021-11-12 18:59:41 RECENT-SEARCH :: DownloadStation: Unable to send torrent to client</pre>

Additional notes:
This seems to be a known issue regarding the depreciation of an API in Synology DS, a new version of the API exists, please see for specs of the new API https://github.com/seansfkelley/nas-download-manager/issues/177#issuecomment-822699863

Workaround:
Downgrade Synology DS to 3.8.15-3563 (impossible on DSM 7.0)

Cheers,

**Comment by JackDandy** [comment]
This is insane, we only just upgraded to Syno this year, are you sure that this isnt some other issue?

**Comment by JackDandy** [comment]
This is insane, we only just upgraded to a "_new_" Syno API v7 in July 2021 - tested on a live machine that was set up in as a Syno VM.

Are you sure you haven't some configuration issue like not using admin account or bad share or something?

If you are absolutely positive that there is an issue you want us to look at, then please can you supply access to such a machine so we can test a fix against? Maybe you can come to IRC to PM me the access details. If you cannot supply access, then we cannot fix.

**Comment by Namlaz** [comment]
Indeed this is insane, geez sorry, couldn't find that you already did the upgrade of the API in my search on the releases. Found the change now indeed on 0.24.9
I'm not using an admin account but one dedicated and limited to DownloadStation, it worked flawlessly for as far as I know and started to fail not so long ago, it seemed to me that it was linked to a DSM 7.0 upgrade - and thus DownloadStation - I just did on the NAS.

As for the "misleading" error : my NAS model is limited to 80 download tasks, which doesn't include seed tasks (370+ active seed tasks were never an issue) and only 9 download tasks are active at the moment, far from the "max number of tasks limit".

I tried to manually log in to the account for the first time post-upgrade, maybe it needed to do some tasks on it to unlock API requests to DS.
Will see tomorrow after the daily backlog search if the error persists.
